ui/src/core/xibo-cms.js

```diff
@@ -1463,6 +1465,17 @@ function dataTableDraw(e, settings) {
         $folderController.removeClass('d-none').addClass('d-inline-flex');
     }
 
+    // Move and show map button inside of the table container
+    if ($mapController.length > 0 && target.closest(".dataTables_wrapper").find('.dataTables_folder .map-controller').length == 0) {
```

Comment:
Mauro will know better, but I think this could be handled as a custom callback to the dataTableDraw function.

Reply:
Israt and I were a bit confused about the dataTableDraw callback, the new buttons will have the same behaviour as the Folder one, so maybe this is the best approach?

Unfortunately with Datatables, we do need to create them outside the table container and move them inside after the draw

Reply:
As this is the only page which has a map currently, it may be more elegant to change the signature of the `dataTableDraw` function so that it can take an optional callback.

We can then pass in a callback only on the displays page, to do the map jiggle wiggle.

```
function dataTableDraw(e, settings, callBack) {
  if (callBack && callBack instanceof Function) {
    callBack();
  }
}```

Reply:
I see, that makes total sense! The `if` statement would prevent the code to run, but we would still be checking for those in all the other pages indeed